website/src/components/TutorialComponents/CodeBlock.tsx

```diff
@@ -121,19 +123,62 @@ function CodeEditor(props: {
         setTmpCode(code.slice()); // use copy not reference
         setCode(props.code);
         setDisabled(true);
+        setHasFocus(false);
         setAllowUndo(true);
         setTimeout(() => {
             setDisabled(props.disabled);
+            setHasFocus(true);
             setAllowUndo(false);
         }, 3000);
     }
 
     const onClickUndo = () => {
-        console.log(tmpCode)
         setCode(tmpCode);
     }
 
-    //   prismIncludeLanguages(Prism);
+    const handleFocus = (e: React.FocusEvent) => {
+        const selectObj = window.getSelection();
+        if (selectObj.rangeCount === 0) {
+            // when focusing on the editor without using the mouse, 
+            // merely from the Tab key
+            const range = new Range();
+            range.collapse(true);
+            selectObj.addRange(range);
+        }
+        setHasFocus(true);
+    }
+
+    const handleBlur = (e: React.FocusEvent) => {
+        setHasFocus(false);
+    }
+
+    const handleKeydown = (e: KeyboardEvent) => {
+        const editor = editorRef.current;
+        if (e.key === 'Escape' && !props.disabled) {
+            if (e.target === editor) {
+                e.stopImmediatePropagation();
+                e.stopPropagation();
+                e.preventDefault();
+                editor.focus(); // sometimes blur won't fire without focus first
+                editor.blur();
+            }
+        }
+    }
+
+
+
+    useEffect(() => {
+        document.addEventListener('keydown',
+            (e) => handleKeydown(e),
```

Comment:
you are creating a new closure with this function call; so this defeats removeEventListener as it won't be the same closure instance you are trying to remove